src/Symfony/Component/HttpKernel/HttpCache/HttpCache.php

```diff
@@ -160,7 +160,11 @@ public function getKernel()
      */
     public function getSurrogate()
     {
-        return $this->getEsi();
+        if (!$this->surrogate instanceof Esi) {
+            throw new \LogicException('This instance of HttpCache was not set up to use ESI as surrogate handler. You must overwrite and use createSurrogate');
+        }
```

Comment:
I might be missing something, but does it really make sense to have this condition in `getSurrogate()` which is suppose to work with any surrogate?